error-handling/readme.md

```diff
@@ -0,0 +1,83 @@
+# Error Handling
+
+## Learning Objectives
+
+```objectives
+- [ ] Give examples of good and bad error messages
+- [ ] Identify appropriate points in code to check for errors
+```
+
+## Understanding What Went Wrong
+
+We have already seen examples of developer questions which can help us fix problems in our code:
+
+1. What I _did_
+2. What I _expected_
+3. What actually _happened_
+
+The problem is that figuring out what happened can be really tricky if the developer doesn't do a good job of **error handling**. In this workshop we're going to discuss error messages, why they are important and what makes some more helpful than others.
+
+What makes an error message "good" depends on who it is intended for. For example, the runtime errors you may have seen while working through the exercises are great for developers while they're working on the code but they aren't very useful for the end user looking at an app on their phone. Knowing the most appropriate way to communicate problems is a key skill for a developer.
+
+### Setup
+
+- **Trainees**: Go to the `error-handling` folder in the [Workshops repo](https://github.com/CodeYourFuture/CYF-Workshops/), navigate to the `webpage` folder and open `index.html` in a browser.
+- **Trainees**: Work in groups of 2 or 3. 
+- **Trainees**: Nominate a member of the group to take notes.
+- **Volunteers**: Nominate someone to facilitate the group discussion. This could be a volunteer or a trainee.
+
+<!--{{<note type="activity" title="Evaluating Error Messages, 15m" >}}-->
+
+Open your web browser and take a look at the web page you just opened. You should see a text box with five buttons below it. When you type a message it should appear below the box. Try it out by typing "hello world" and see what happens. 
+
+The problem we have is that none of the buttons seem to be working! They all have the same problem but they aren't all telling us the same thing. We could probably open `script.js` and easily figure out where the problem is, but this won't be an option for the end users of our code.
+
+In your groups, **without looking at the code**, answer the following questions for each button:
+
+1. Can you find where the error message is being displayed? (Hint: there may not always be an error message)
```

Comment:
I would swap buttons 1 and 2 so that the _first_ button they click isn't an unending "there's no error message" investigation.